
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Documentation for the Progenetix oncogenomics resource">
      
      
        <meta name="author" content="Michael Baudis">
      
      
      <link rel="icon" href="../img/site-logo-topright.png">
      <meta name="generator" content="mkdocs-1.3.0, mkdocs-material-8.3.5">
    
    
      
        <title>Use Case Examples - Progenetix Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.4a0965b7.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
          
          
          <meta name="theme-color" content="#ff6e42">
        
      
      
  
  
    
    
  
  
  <style>:root{--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3.604 3.089A.75.75 0 0 1 4 3.75V8.5h.75a.75.75 0 0 1 0 1.5h-3a.75.75 0 1 1 0-1.5h.75V5.151l-.334.223a.75.75 0 0 1-.832-1.248l1.5-1a.75.75 0 0 1 .77-.037zM8.75 5.5a.75.75 0 0 0 0 1.5h11.5a.75.75 0 0 0 0-1.5H8.75zm0 6a.75.75 0 0 0 0 1.5h11.5a.75.75 0 0 0 0-1.5H8.75zm0 6a.75.75 0 0 0 0 1.5h11.5a.75.75 0 0 0 0-1.5H8.75zM5.5 15.75c0-.704-.271-1.286-.72-1.686a2.302 2.302 0 0 0-1.53-.564c-.535 0-1.094.178-1.53.565-.449.399-.72.982-.72 1.685a.75.75 0 0 0 1.5 0c0-.296.104-.464.217-.564A.805.805 0 0 1 3.25 15c.215 0 .406.072.533.185.113.101.217.268.217.565 0 .332-.069.48-.21.657-.092.113-.216.24-.403.419l-.147.14c-.152.144-.33.313-.52.504l-1.5 1.5a.75.75 0 0 0-.22.53v.25c0 .414.336.75.75.75H5A.75.75 0 0 0 5 19H3.31l.47-.47c.176-.176.333-.324.48-.465l.165-.156a5.98 5.98 0 0 0 .536-.566c.358-.447.539-.925.539-1.593z"/></svg>');}</style>

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../css/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="deep-orange" data-md-color-accent="">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#explore-gene-cnvs" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Progenetix Documentation" class="md-header__button md-logo" aria-label="Progenetix Documentation" data-md-component="logo">
      
  <img src="../img/site-logo-topright.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Progenetix Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Use Case Examples
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/progenetix/progenetix-web" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"/></svg>
  </div>
  <div class="md-source__repository">
    progenetix-web
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  

<nav class="md-nav md-nav--primary md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Progenetix Documentation" class="md-nav__button md-logo" aria-label="Progenetix Documentation" data-md-component="logo">
      
  <img src="../img/site-logo-topright.png" alt="logo">

    </a>
    Progenetix Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/progenetix/progenetix-web" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"/></svg>
  </div>
  <div class="md-source__repository">
    progenetix-web
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Documentation Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../news/" class="md-nav__link">
        News & Changes
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../ui/" class="md-nav__link">
        Pages & Forms
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../services/" class="md-nav__link">
        Services & API
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Use Case Examples
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Use Case Examples
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#explore-gene-cnvs" class="md-nav__link">
    Explore Gene CNVs
  </a>
  
    <nav class="md-nav" aria-label="Explore Gene CNVs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-procedure" class="md-nav__link">
    Example Procedure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualize-cnv-patterns" class="md-nav__link">
    Visualize CNV patterns
  </a>
  
    <nav class="md-nav" aria-label="Visualize CNV patterns">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-procedure_1" class="md-nav__link">
    Example Procedure
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-or-plot-cnv-frequencies" class="md-nav__link">
    Download or Plot CNV Frequencies
  </a>
  
    <nav class="md-nav" aria-label="Download or Plot CNV Frequencies">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#collation-plots" class="md-nav__link">
    Collation plots
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-based-histograms" class="md-nav__link">
    Query-based histograms
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-procedure-download-cnv-frequencies" class="md-nav__link">
    Example Procedure - Download CNV Frequencies
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-procedure-download-or-embed-cnv-frequency-plot" class="md-nav__link">
    Example Procedure - Download or embed CNV frequency plot
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#download-sample-data" class="md-nav__link">
    Download Sample Data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pgxrpi-an-r-library-to-access-progenetix-data" class="md-nav__link">
    pgxRpi, an R Library to Access Progenetix Data
  </a>
  
    <nav class="md-nav" aria-label="pgxRpi, an R Library to Access Progenetix Data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-biosample-information" class="md-nav__link">
    Retrieve biosample information
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-cnv-coverage-data-of-biosamples-from-specific-cohorts" class="md-nav__link">
    Query CNV coverage data of biosamples from specific cohorts
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-and-export-segment-copy-number-variant-data" class="md-nav__link">
    Query and export segment copy number variant data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#query-and-visualize-cnv-frequencies" class="md-nav__link">
    Query and visualize CNV frequencies
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#user-provided-cnv-data" class="md-nav__link">
    User-Provided CNV Data
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../classifications-and-ontologies/" class="md-nav__link">
        Classifications, Ontologies & Standards
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../publication-collection/" class="md-nav__link">
        Publication Collection
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../progenetix-data-review/" class="md-nav__link">
        Data Review
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../beaconplus/" class="md-nav__link">
        Beacon+ & bycon
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../technical-notes/" class="md-nav__link">
        Technical Notes
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="http://progenetix.org" class="md-nav__link">
        Progenetix Data
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="http://info.baudisgroup.org" class="md-nav__link">
        Baudisgroup @ UZH
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/progenetix/progenetix-web/edit/main/docs/use-cases.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="use-cases">Use Cases<a class="headerlink" href="#use-cases" title="Permanent link">&para;</a></h1>
<h2 id="explore-gene-cnvs">Explore Gene CNVs<a class="headerlink" href="#explore-gene-cnvs" title="Permanent link">&para;</a></h2>
<p>One of the main use cases for the Progenetix resource is the exploration of frequency and disease specificity of genes of interest. Traditionally, the relevance of somatic CNVs hitting a gene in the context of cancer are judged by</p>
<ul>
<li>the CNV frequency (i.e. in what fraction of samples the a CNV in this gene is being observed)</li>
<li>the relative specificity, i.e. how CNVs in this gene compare to<ul>
<li>the overall amount of CNVs in the samples</li>
<li>the local specificity, i.e. the "focality" of the CNVs</li>
</ul>
</li>
</ul>
<p>The <a href="https://progenetix.org/biosamples/">Progenetix Search Page</a> supports the exploration of regional CNVs through</p>
<ul>
<li>support for inserting positions of genes or cytobands into standard Beacon query parameter fields</li>
<li>selection support for hierarchical disease annotations</li>
<li>providing example queries as templates</li>
</ul>
<p>The response of the queries for genomic variants provide some basic statistics, e.g. the relative frequencies of these variants for each classification code (e.g. NCIT, ICD-O) encountered in teh matched samples.</p>
<h3 id="example-procedure">Example Procedure<a class="headerlink" href="#example-procedure" title="Permanent link">&para;</a></h3>
<p><strong>TBD</strong></p>
<hr />
<h2 id="visualize-cnv-patterns">Visualize CNV patterns<a class="headerlink" href="#visualize-cnv-patterns" title="Permanent link">&para;</a></h2>
<p>The Progenetix resource contains pre-computed CNV frequency profiles for all diagnostic
and technical (e.g. derived from a publication or technical platform) categories. These
profiles provide a rapid overview of CNV events along the genome but also can serve for
the exploration of possible genomic subsets hidden behind the overview profiles.</p>
<h4 id="cnv-profile-clustering">CNV profile clustering<a class="headerlink" href="#cnv-profile-clustering" title="Permanent link">&para;</a></h4>
<h3 id="example-procedure_1">Example Procedure<a class="headerlink" href="#example-procedure_1" title="Permanent link">&para;</a></h3>
<p><strong>TBD</strong></p>
<hr />
<h2 id="download-or-plot-cnv-frequencies">Download or Plot CNV Frequencies<a class="headerlink" href="#download-or-plot-cnv-frequencies" title="Permanent link">&para;</a></h2>
<h3 id="collation-plots">Collation plots<a class="headerlink" href="#collation-plots" title="Permanent link">&para;</a></h3>
<p>The Progenetix resource provides pre-computed CNV frequencies for all its
"collations" such as</p>
<ul>
<li>cancer types by e.g. NCIt, ICD-O morphology and topography codes</li>
<li>experimental series, e.g. all samples from a given publication</li>
<li>custom cohorts, e.g. all samples used in a Progenetix meta-analysis or
external project such as TCGA</li>
</ul>
<p>This data can be accessed through the Progenetix API in data and image format.</p>
<p><img alt="Example CNV histogram with custom parameters" src="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22" /></p>
<p>Interval frequencies are per default stored in a 1Mb binned format. More
information about the API use can be found <a href="../services/#pgxseg-segment-cnv-frequencies">in the IntervalFrequencies API documentation</a>.</p>
<h3 id="query-based-histograms">Query-based histograms<a class="headerlink" href="#query-based-histograms" title="Permanent link">&para;</a></h3>
<p>The Progenetix Beacon responses - depending on their type - usually contain a <em>handover</em>
URL to retrieve CNV histogram and/or sample plots of the samples matched by the query.
The <code>bycon</code> API now offers a direct access to the histograms without the need to deparse
JSON response first. The switch to the histogram is ionitiated by adding <code>&amp;output=histoplot</code>
to the Beacon query URL. Then, the API will first query the samples and then perfor
a handover to the plotting API. Please be aware that this procedure is best suited for limited
queries and may lead to a time-out.</p>
<h4 id="examples">Examples:<a class="headerlink" href="#examples" title="Permanent link">&para;</a></h4>
<p><img alt="Example query-based CNV histogram" src="https://progenetix.org/beacon/biosamples/?referenceName=9&amp;variantType=EFO:0030067&amp;start=21500000&amp;start=21975098&amp;end=21967753&amp;end=22500000&amp;filters=NCIT:C3058&amp;filters=pgx:cohort-celllines&amp;output=histoplot" /></p>
<ul>
<li><a href="https://progenetix.org/beacon/biosamples/?referenceName=9&amp;variantType=EFO:0030067&amp;start=21500000&amp;start=21975098&amp;end=21967753&amp;end=22500000&amp;filters=NCIT:C3058&amp;filters=pgx:cohort-celllines&amp;output=histoplot">progenetix.org/beacon/biosamples/?referenceName=9&amp;variantType=EFO:0030067&amp;start=21500000&amp;start=21975098&amp;end=21967753&amp;end=22500000&amp;filters=NCIT:C3058&amp;filters=pgx:cohort-celllines&amp;output=histoplot</a><ul>
<li>a search for samples with focal deletion in the <em>CDKN2A</em> locus, limited to glioblastoma cell lines </li>
</ul>
</li>
</ul>
<h3 id="example-procedure-download-cnv-frequencies">Example Procedure - Download CNV Frequencies<a class="headerlink" href="#example-procedure-download-cnv-frequencies" title="Permanent link">&para;</a></h3>
<p>Typical cases for the use of collation-specific frequency data could be e.g.
the visualization of CNV tracks with 3rd party tools such as <a href="http://www.circos.ca/software/">Circos</a>
or integration in data analysis workflows, e.g. for comparing target genes to
local, disease-specific CNV frequencies.</p>
<h4 id="getting-cancer-type-cnv-frequencies">Getting cancer type CNV frequencies<a class="headerlink" href="#getting-cancer-type-cnv-frequencies" title="Permanent link">&para;</a></h4>
<p>All cancer codes for a given classification system can be retrieved though:</p>
<ul>
<li>NCIT<ul>
<li><a href="https://progenetix.org/services/collations?filters=NCIT&amp;output=text">progenetix.org/services/collations?filters=NCIT&amp;output=text</a></li>
</ul>
</li>
<li>ICD-O Morphologies<ul>
<li><a href="https://progenetix.org/services/collations?filters=icdom&amp;method=counts&amp;output=text">progenetix.org/services/collations?filters=icdom&amp;method=counts&amp;output=text</a></li>
<li>please be aware that we have to use transformed ICD-O codes; e.g.
  "ICD-O 8500/3" is represented as <code>pgx:icdom-85003</code> (<code>s/^(\d{4})\/(\d)$/pgx:icdom-$1$2/</code>)</li>
</ul>
</li>
<li>ICD-O Topographies<ul>
<li><a href="https://progenetix.org/services/collations?filters=icdot&amp;method=counts&amp;output=text">progenetix.org/services/collations?filters=icdot&amp;method=counts&amp;output=text</a></li>
</ul>
</li>
</ul>
<h4 id="download-the-data-file">Download the data file<a class="headerlink" href="#download-the-data-file" title="Permanent link">&para;</a></h4>
<p>For any of those codes one can create a <code>.pgxseg</code> file downloader link for the
<a href="../services/#pgxseg-segment-cnv-frequencies">"IntervalFrequencies" service</a>:</p>
<h5 id="examples_1">Examples<a class="headerlink" href="#examples_1" title="Permanent link">&para;</a></h5>
<ul>
<li><a href="https://progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=NCIT:C105555">progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=NCIT:C105555</a></li>
<li><a href="https://progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=icdom-85003">progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=icdom-85003</a></li>
</ul>
<h3 id="example-procedure-download-or-embed-cnv-frequency-plot">Example Procedure - Download or embed CNV frequency plot<a class="headerlink" href="#example-procedure-download-or-embed-cnv-frequency-plot" title="Permanent link">&para;</a></h3>
<p>For the generation of CNV frequency plots, the same procedure as above for
identifying existing frequency maps can be applied. CNV hsitograms in <a href="../services/#image-formats">SVG format</a>
can be generated for download or embedding through a canonical service URL with
added (single) collation code.</p>
<h4 id="additional-plot-parameters">Additional plot parameters<a class="headerlink" href="#additional-plot-parameters" title="Permanent link">&para;</a></h4>
<p>Plot parameters can be added to the request using a standard <code>&amp;-__parameter__=__value__</code>
syntax. Please be aware of the <code>-</code> prefix.</p>
<ul>
<li><code>-size_plotimage_w_px</code><ul>
<li>modifies the width of the plot image in px (default 800)</li>
<li><a href="https://progenetix.org/services/collationPlots/?id=PMID:22824167&amp;-size_plotimage_w_px=1084">progenetix.org/services/collationPlots/?id=PMID:22824167&amp;-size_plotimage_w_px=1084</a></li>
</ul>
</li>
<li><code>-size_plotarea_h_px</code><ul>
<li>height of the plot area (excluding labels etc.) in px (default 100)</li>
<li><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=300">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=300</a></li>
</ul>
</li>
<li><code>-value_plot_y_max</code><ul>
<li>modifies the histogram's maximum value in percent (default 100)</li>
<li><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=pgx:cohort-TCGAcancers&amp;-value_plot_y_max=50">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=pgx:cohort-TCGAcancers&amp;-value_plot_y_max=50</a></li>
</ul>
</li>
<li><code>-colorschema</code><ul>
<li>change of colors used for gains and losses</li>
<li>options<ul>
<li><code>orangeblue</code> (default)</li>
<li><code>redgreen</code></li>
<li><code>greenred</code></li>
<li><code>bluered</code></li>
</ul>
</li>
</ul>
</li>
<li>
<p><code>-chr2plot</code></p>
<ul>
<li>comma-concatenated list of chromosomes to plot</li>
<li>
<p>default is 1 -&gt; 22 since X &amp; Y are not always correctly normalized for CNV
  frequencies</p>
<ul>
<li><code>-chr2plot=1,2,3,44,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y</code></li>
<li><code>-chr2plot=9</code></li>
</ul>
</li>
<li>
<p><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22</a></p>
<ul>
<li>see example above (live representation of embedded API call for this example)</li>
</ul>
</li>
</ul>
</li>
</ul>
<h5 id="examples_2">Examples<a class="headerlink" href="#examples_2" title="Permanent link">&para;</a></h5>
<ul>
<li><a href="https://progenetix.org/services/collationPlots/?id=pgx:icdom-85003">progenetix.org/services/collationPlots/?id=pgx:icdom-85003</a></li>
<li><a href="https://progenetix.org/services/collationPlots/?id=PMID:22824167">progenetix.org/services/collationPlots/?id=PMID:22824167</a></li>
</ul>
<hr />
<h2 id="download-sample-data">Download Sample Data<a class="headerlink" href="#download-sample-data" title="Permanent link">&para;</a></h2>
<p>The sample annotations for any type of query - particularly but not limited to
single identifier values - can be downloaded in either <a href="../beaconplus/#beacon-api">Beacon v2 JSON</a>
or in a tab-delimited text format (<code>&amp;output=table</code>).</p>
<h5 id="examples_3">Examples<a class="headerlink" href="#examples_3" title="Permanent link">&para;</a></h5>
<ul>
<li>Download all TCGA cancer samples from Progenetix as tab-delimited table<ul>
<li><a href="http://progenetix.org/beacon/biosamples/?filters=pgx:cohort-TCGAcancers&amp;output=table">progenetix.org/beacon/biosamples/?filters=pgx:cohort-TCGAcancers&amp;output=table</a></li>
</ul>
</li>
</ul>
<hr />
<h2 id="pgxrpi-an-r-library-to-access-progenetix-data">pgxRpi, an R Library to Access Progenetix Data<a class="headerlink" href="#pgxrpi-an-r-library-to-access-progenetix-data" title="Permanent link">&para;</a></h2>
<p><code>pgxRpi</code> is an API wrapper package to access data from Progenetix database. More details about this package are in the <a href="https://github.com/progenetix/pgxRpi">vignettes</a>. There are several functions in this R package.</p>
<h3 id="retrieve-biosample-information">Retrieve biosample information<a class="headerlink" href="#retrieve-biosample-information" title="Permanent link">&para;</a></h3>
<p>You can select biosamples from specific groups of interests, chosen by a filter. The description about <em>filters</em> is <a href="https://docs.progenetix.org/classifications-and-ontologies/">here</a>.</p>
<p><div class="highlight"><pre><span></span><code>biosamples &lt;- pgxLoader(type=&quot;biosample&quot;, filters = &quot;NCIT:C3512&quot;,codematches = TRUE)
</code></pre></div>
The returned biosample information includes biosample id, various codes for tumor types, tumor stage, survival data, associated literature or research project, etc.</p>
<h3 id="query-cnv-coverage-data-of-biosamples-from-specific-cohorts">Query CNV coverage data of biosamples from specific cohorts<a class="headerlink" href="#query-cnv-coverage-data-of-biosamples-from-specific-cohorts" title="Permanent link">&para;</a></h3>
<p>The coverage is calculated across 1MB genomic bins, chromosomal arms, whole chromosomes, or whole genome.</p>
<p>The CNV coverage across genomic bins can be accessed by setting <code>output</code> = "pgxmatrix". More details about the data format "pgxmatrix" see the <a href="https://docs.progenetix.org/services/#cnv-status-matrix">documentation</a>.</p>
<div class="highlight"><pre><span></span><code>cnv.status &lt;- pgxLoader(type=&quot;variant&quot;, filters = &quot;NCIT:C3058&quot;, output=&quot;pgxmatrix&quot;, codematches = T)
</code></pre></div>
<p>The CNV coverage across chromosomal arms, chromosomes, or whole genome can be accessed by setting <code>output</code> = "coverage".</p>
<div class="highlight"><pre><span></span><code>cnv.status &lt;- pgxLoader(type=&quot;variant&quot;, filters = &quot;NCIT:C4443&quot;, output=&quot;coverage&quot;, codematches = F)
</code></pre></div>
<h3 id="query-and-export-segment-copy-number-variant-data">Query and export segment copy number variant data<a class="headerlink" href="#query-and-export-segment-copy-number-variant-data" title="Permanent link">&para;</a></h3>
<p>You can download the copy number variant data of individual biosamples. The biosample id can be queried by pgxRpi or by Progenetix <a href="https://progenetix.org/biosamples/">website</a>.
The variant data exportation supports different output formats, more information see vignettes.</p>
<div class="highlight"><pre><span></span><code>variants &lt;- pgxLoader(type=&quot;variant&quot;, biosample_id = c(&quot;pgxbs-kftva6du&quot;,&quot;pgxbs-kftva6dv&quot;,&quot;pgxbs-kftva6dx&quot;),output = &quot;pgxseg&quot;)
</code></pre></div>
<h3 id="query-and-visualize-cnv-frequencies">Query and visualize CNV frequencies<a class="headerlink" href="#query-and-visualize-cnv-frequencies" title="Permanent link">&para;</a></h3>
<p>You can query the CNV frequency of specific filters, namely specific cohorts. There are two available data formats. One is <a href="https://docs.progenetix.org/services/#pgxseg-segment-cnv-frequencies"><code>.pgxseg</code></a>, good for visualization. Another is <a href="https://docs.progenetix.org/services/#cnv-frequency-matrix"><code>.pgxmatrix</code></a>, good for analysis.</p>
<div class="highlight"><pre><span></span><code>frequency &lt;- pgxLoader(type=&quot;frequency&quot;, output =&#39;pgxseg&#39;,
                         filters=c(&quot;NCIT:C4038&quot;,&quot;pgx:icdom-85003&quot;), 
                         codematches = TRUE)
</code></pre></div>
<p>The data visualization requires the input data with <code>.pgxseg</code> format. You can plot the frequency by genome, by chromosomes, or plot like circos.</p>
<div class="highlight"><pre><span></span><code>pgxFreqplot(frequency, filters=&#39;pgx:icdom-85003&#39;)
</code></pre></div>
<p><img src="../img/pgxRpi-freq-plot-by-genome.png" style="margin-left: auto; margin-right:auto" /></p>
<div class="highlight"><pre><span></span><code>pgxFreqplot(frequency, filters=&#39;NCIT:C4038&#39;,chrom=c(1,2,3), layout = c(3,1))
</code></pre></div>
<p><img src="../img/pgxRpi-freq-plot-by-chrom.png" style="margin-left: auto; margin-right:auto" /></p>
<div class="highlight"><pre><span></span><code>pgxFreqplot(frequency,filters= c(&quot;NCIT:C4038&quot;,&quot;pgx:icdom-85003&quot;),circos = TRUE) 
</code></pre></div>
<p><img src="../img/pgxRpi-freq-plot-by-circos.png" style="margin-left: auto; margin-right:auto" /></p>
<hr />
<h2 id="user-provided-cnv-data">User-Provided CNV Data<a class="headerlink" href="#user-provided-cnv-data" title="Permanent link">&para;</a></h2>
<p>The Progenetix resource has a limited option to visualize CNV data provided by the users. Data has to be formatted in a standard tab-delimited columnar format, preferably using the <a href="../services/#data-file-formats-pgxseg-segments"><code>.pgxseg</code> file format</a>. Additional information can be found on the upload service page:</p>
<ul>
<li><a href="https://progenetix.org/service-collection/uploader/">Upload Files for CNV Visualization</a></li>
</ul>
<p>The data is only temporary stored on the server with a randomiized access path. At this time we do not provide any long term or login protected data storage and do not store access data.</p>
<h4 id="example-procedure_2">Example Procedure<a class="headerlink" href="#example-procedure_2" title="Permanent link">&para;</a></h4>
<p><strong>TBD</strong></p>
<hr />

              
            </article>
            
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../services/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Services &amp; API" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Services & API
            </div>
          </div>
        </a>
      
      
        
        <a href="../classifications-and-ontologies/" class="md-footer__link md-footer__link--next" aria-label="Next: Classifications, Ontologies &amp; Standards" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Classifications, Ontologies & Standards
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      &copy; Copyright 2022, Michael Baudis and progenetix.org
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
      
      
    
    <a href="https://twitter.com/progenetix" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg>
    </a>
  
    
    
      
      
    
    <a href="https://github.com/progenetix" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link", "search.highlight", "search.share", "navigation.instant", "navigation.tracking", "navigation.sections", "toc.integrate"], "search": "../assets/javascripts/workers/search.85cb4492.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.431f3d41.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/tablesort/5.2.1/tablesort.min.js"></script>
      
    
  </body>
</html>